### ISSUE TITLE
Plz help

### ISSUE THREAD

**Issue by Affanabbbas** [IrohXu/ViTASD#11]
pytorch_lightning.utilities.exceptions.MisconfigurationException: You requested gpu: [0, 1, 2, 3]
 But your machine only has: [0]

I am running the code on google colab. Is the issue related to the need of physical GPU?

**Comment by IrohXu** [comment]
https://github.com/PediaMedAI/ViTASD/blob/f96561962dbae63fc1e168c081946819e6262e9b/configs/config_vitasd_large.yaml#L26

Set here to 1 may help.

**Comment by Affanabbbas** [comment]
/usr/local/lib/python3.10/dist-packages/pytorch_lightning/callbacks/model_checkpoint.py:616: UserWarning: Checkpoint directory /content/ViTASD/lightning_logs/AffectNet-B exists and is not empty.
  rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

pytorch_lightning.utilities.exceptions.MisconfigurationException: The provided lr scheduler `CosineLRScheduler` doesn't follow PyTorch's LRScheduler API. You should override the `LightningModule.lr_scheduler_step` hook with your own logic if you are using a custom LR scheduler


Thanks for your reply. I resolved the previous issue but got a new one as said above. plz tell solution to that.

**Comment by Affanabbbas** [comment]
I am still waiting for your reply brother.

**Comment by IrohXu** [comment]
It seems that it is a version error. Try to downgrade `pytorch_lightning` will solve it.
